splash/browser_tab.py

```diff
@@ -827,11 +839,21 @@ def get(self, url, callback, headers=None, follow_redirects=True):
 
     def _send_request(self, url, callback, method='GET', body=None,
                       headers=None):
-        # XXX: The caller must ensure self._delete_reply is called in a callback.
-        if method != 'GET':
-            raise NotImplementedError()
-        request = self.request_obj(url, headers=headers)
-        reply = self.network_manager.get(request)
+        # XXX: The caller must ensure self._delete_reply is called in a callback
+        request = self.request_obj(url, headers=headers, body=body)
+
+        if method.upper() not in OPERATION_QT_CONSTANTS:
+            raise ValueError("Invalid HTTP method: '{}'".format(method))
+
+        operation = getattr(self.network_manager, method.lower())
```

Comment:
I think this is not correct: there is no .options method, and a method for DELETE is called `deleteResource`, not 'delete`. Maybe we can use [sendCustomRequest](http://doc.qt.io/qt-5/qnetworkaccessmanager.html#sendCustomRequest) instead?